### ISSUE TITLE
ending debug task from within IntelliJ hangs the process and locks the files

### ISSUE THREAD

**Issue by Silthus** [spigradle/spigradle#16]
When I run the `debugSpigot` task and then "end/kill" it from within IntelliJ it leaves the process open and locks the plugin JARs.

![image](https://user-images.githubusercontent.com/755327/86291790-d696b400-bbef-11ea-8d43-d5bbbf538bc4.png)

I need to go into the taskmanager and kill the open JVM process for the lock to be released. Before that I cannot delete or update the plugins in the debug/plugins folder.

Maybe there is a way to gracefully end the task if it gets a exit code?

**Comment by EntryPointKR** [comment]
Hmm. previously I also had this problem, but now weirdly cannot reproduce it.
Moreover, I'm not sure able to fix it even If I get to reproduce because it seems just disconnecting from IntelliJ and the task still being executed.

Is the process still kept even you clicked the red square button (not the X button) you specified on the image?
A suggestion is, terminate it using 'stop' command for the graceful shutdown. and a programmatic solution is, execute the jar as a new process/cmd-window instead of the Gradle's JavaExec task. I'll consider it.

**Comment by Silthus** [comment]
Mmh I can't reproduce it now either. Strange...

I also got a workaround, at least with IntelliJ. The "workaround" also enables me to do a one click server debugging and can hit breakpoints for plugins in onEnable without getting stressed to attach the debugger.

I use the following IntelliJ run configuration:

```xml
<component name="ProjectRunConfigurationManager">
  <configuration default="false" name="Server" type="JarApplication">
    <option name="JAR_PATH" value="$PROJECT_DIR$/debug/spigot/server.jar" />
    <option name="VM_PARAMETERS" value="-Xmx1G -Xms1G -Dfile.encoding=UTF-8" />
    <option name="PROGRAM_PARAMETERS" value="--nogui --log-strip-color" />
    <option name="WORKING_DIRECTORY" value="C:\develop\Minecraft\templates\Spigot-Workspace-Template\debug\spigot" />
    <option name="ALTERNATIVE_JRE_PATH_ENABLED" value="true" />
    <option name="ALTERNATIVE_JRE_PATH" value="1.8" />
    <module name="spigot-plugin-template.main" />
    <method v="2">
      <option name="Gradle.BeforeRunTask" enabled="true" tasks="prepareSpigot" externalProjectPath="$PROJECT_DIR$" vmOptions="" scriptParameters="" />
      <option name="Gradle.BeforeRunTask" enabled="true" tasks="prepareSpigotPlugins" externalProjectPath="$PROJECT_DIR$" vmOptions="" scriptParameters="" />
    </method>
  </configuration>
</component>
```

**Comment by EntryPointKR** [comment]
That's really nice. if we want to run Paper, the `downloadPaper` can be used as a replacement of `prepareSpigot`.
I'll automate it to generate in IDEA support.

**Comment by EntryPointKR** [comment]
I tried to implement it, but unfortunately, there's no particular support for the RunConfiguration `JarApplication` in Gradle plugin `idea-ext-plugin`. 
If I handle the raw XML, I can achieve it but don't want to do it. But, there's the `Application` - can we execute the jar using it? It seems to require `Main class`.

**Comment by Silthus** [comment]
Mmh I am not sure either. Maybe fork the `idea-ext-plugin` and add the required task?

**Comment by EntryPointKR** [comment]
Yeah, I opened a pull request for the `JarApplication` generation. If it merged, maybe this feature becomes on the next version of IDEA.

**Comment by EntryPointKR** [comment]
Good news, the PR https://github.com/JetBrains/intellij-community/pull/1405 was merged.
Now I can implement the `JarApplication` generation :)

**Comment by EntryPointKR** [comment]
This will be available on the 2020.2.x IDEA.

**Comment by EntryPointKR** [comment]
Now available on the 2020.3!
